frontend/packages/kubevirt-plugin/src/components/cdi-upload-provider/upload-pvc-form/upload-pvc-form.tsx

```diff
@@ -167,26 +181,42 @@ export const UploadPVCForm: React.FC<UploadPVCFormProps> = (props) => {
     setAccessMode(event.currentTarget.value);
   };
 
-  const onlyPvcSCs = React.useCallback((sc: StorageClass) => !isObjectSC(sc), []);
-
-  const operatingSystems = getTemplateOperatingSystems(commonTemplates);
-
-  const handleOs = (osKey) => {
-    const operatingSystem = operatingSystems.find((newOs) => newOs.id === osKey);
+  const handleGoldenCheckbox = (checked) => {
+    setIsGolden(checked);
+    if (checked) {
+      const operatingSystem = operatingSystems.find((newOs) => newOs.id === os);
+      setNamespace(operatingSystem?.dataVolumeNamespace || TEMPLATE_VM_GOLDEN_OS_NAMESPACE);
+      if (pvcName && !os) {
+        setPvcName('');
+      } else {
+        setPvcName(operatingSystem?.dataVolumeName || os);
+      }
+    }
+    if (!checked) {
+      setNamespace(props.namespace);
+    }
+  };
 
-    setOs(osKey);
-    setPvcName(operatingSystem?.dataVolumeName || osKey);
-    setNamespace(operatingSystem?.dataVolumeNamespace || TEMPLATE_VM_GOLDEN_OS_NAMESPACE);
+  const handleOs = (newOs) => {
+    setOs(newOs);
+    const operatingSystem = operatingSystems.find((o) => o.id === newOs);
+    setPvcName(operatingSystem?.dataVolumeName || newOs);
   };
 
+  const onlyPvcSCs = React.useCallback((sc: StorageClass) => !isObjectSC(sc), []);
+
   React.useEffect(() => {
-    if (isGolden && pvcName && !os) {
-      setPvcName('');
-    }
-    if (!isGolden) {
-      setNamespace(props.namespace);
+    if (!isDisabled && !updatedPVCName.current) {
```

Comment:
we probably would not need the ref if we depended just on isLoaded and not generice isDisabled